src/System.Threading/tests/ReaderWriterLockTests.cs

```diff
@@ -744,6 +744,31 @@ public static void AtomicDowngradeTest()
             trwl.Dispose();
         }
 
+        [Fact]
+        public static void OverflowOnExcessiveNestedReaderLock()
+        {
+            ReaderWriterLock rwl = new ReaderWriterLock();
+            for (int i = 0; i != ushort.MaxValue; ++i)
```

Comment:
A couple of milliseconds. Since all the locks are from the same thread with no contention they each don't take very long. The `RemovingWaitingParticipants` test is a bit worse because it has to have another thread signal the barrier and not finish the test until it has done so, and experimenting to neither waste too much time waiting nor spin too much found that it takes around 100-200ms on my machine running a similar method outside of xUnit. I don't know whether xUnit schedules in such as to give the thread that had been running the test extra work.